src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,715 @@
+                                                    ov::loaded_from_cache};
+        return ro_properties;
+    };
+    const auto& to_string_vector = [](const std::vector<ov::PropertyName>& properties) {
+        std::vector<std::string> ret;
+        for (const auto& property : properties) {
+            ret.emplace_back(property);
+        }
+        return ret;
+    };
+
+    if (ov::supported_properties == name) {
+        auto supported_properties = default_ro_properties();
+        add_ro_properties(ov::supported_properties.name(), supported_properties);
+        add_ro_properties(ov::device::properties.name(), supported_properties);
+        add_ro_properties(ov::device::priorities.name(), supported_properties);
+        return decltype(ov::supported_properties)::value_type(supported_properties);
+    } else if (EXEC_NETWORK_METRIC_KEY(SUPPORTED_METRICS) == name) {
+        auto metrics = default_ro_properties();
+        add_ro_properties(METRIC_KEY(SUPPORTED_METRICS), metrics);
+        add_ro_properties(METRIC_KEY(SUPPORTED_CONFIG_KEYS), metrics);
+        return to_string_vector(metrics);
+    } else if (EXEC_NETWORK_METRIC_KEY(SUPPORTED_CONFIG_KEYS) == name) {
+        return to_string_vector(m_cfg.get_supported());
+    } else if (ov::device::properties == name) {
+        ov::AnyMap all_devices = {};
+        for (auto&& comp_model_desc : m_compiled_submodels)
+            if (all_devices.count(comp_model_desc.device) == 0)
+                all_devices[comp_model_desc.device] =
+                    comp_model_desc.compiled_model->get_property(ov::supported_properties.name());
+        return all_devices;
+    } else if (ov::model_name == name) {
+        return decltype(ov::model_name)::value_type(m_name);
+    } else if (ov::loaded_from_cache == name) {
+        return decltype(ov::loaded_from_cache)::value_type{m_loaded_from_cache};
+    } else if (ov::optimal_number_of_infer_requests == name) {
+        unsigned int value = 0u;
+        for (auto&& comp_model_desc : m_compiled_submodels) {
+            value = std::max(value,
+                             comp_model_desc.compiled_model->get_property(ov::optimal_number_of_infer_requests.name())
+                                 .as<unsigned int>());
+        }
+        return decltype(ov::optimal_number_of_infer_requests)::value_type{value};
+    } else if (ov::execution_devices == name) {
+        std::vector<std::string> device_names;
+        std::set<std::string> s;
+        for (auto&& comp_model_desc : m_compiled_submodels) {
+            if (s.count(comp_model_desc.device) != 0)
+                continue;
+            s.insert(comp_model_desc.device);
+            device_names.push_back(comp_model_desc.device);
+        }
+        return decltype(ov::execution_devices)::value_type{device_names};
+    }
+    return m_cfg.get(name);
+}
+
+const std::vector<ov::Output<const ov::Node>>& ov::hetero::CompiledModel::inputs() const {
+    return m_compiled_inputs;
```

Comment:
[nkogteva](https://github.com/nkogteva) [Jul 6, 2023](https://github.com/openvinotoolkit/openvino/pull/18392#discussion_r1254428879)

```
BTW why do we need to override this functions?
```

Member Author
@[vurusovs](https://github.com/vurusovs) vurusovs [Jul 6, 2023](https://github.com/openvinotoolkit/openvino/pull/18392#discussion_r1254479363)
```
In case of API 1.0, AddPreprocessing transformation will be applied on device side and will change parameters of inputs/outputs (e.g. precision). To translate updated parameters on Hetero side, I need to replace original m_inputs/m_outputs with inputs/outputs from already compiled models
```
Member
@[nkogteva](https://github.com/nkogteva) nkogteva [Jul 7, 2023](https://github.com/openvinotoolkit/openvino/pull/18392#discussion_r1255328897)
```
    In case of API 1.0, AddPreprocessing transformation will be applied on device side and will change parameters of inputs/outputs (e.g. precision). To translate updated parameters on Hetero side, I need to replace original m_inputs/m_outputs with inputs/outputs from already compiled models

Inputs and outputs of original model should not be changed. They should the same in base class.
```
Member Author
@[vurusovs](https://github.com/vurusovs) vurusovs [Jul 10, 2023](https://github.com/openvinotoolkit/openvino/pull/18392#discussion_r1258141838)
```
For the moment it isn't possible because of API 1.0 AddPreprocessing that changes inputs/outputs attributes, so Hetero CompiledModel inputs()/outputs() doesn't translate actual information

So current idea is next - replace inputs/outputs from ov::Model with compiled submodels inputs/outputs. In case of API 2.0 there should be no changes by subplugins.
Also it helps with import/export - instead of deserializing inputs/outputs it can be just restored from imported model, so no need to serialize by Hetero plugin
```

Reply:
As far as I understood when all plugins will be moved to API 2.0 it will be not needed. Could you please create a ticket which will depends on appropriate plugins API movement for HETERO cleanup?

Reply:
CVS-115322